### DISCUSSION TITLE
type conversion error in LinuxSerialDriver/LinuxSerialDriver.hpp?

### DISCUSSION THREAD

**Discussion by danjwait** [nasa/fprime#1343] [Q&A] [answered]
Trying to complete the GPS tutorial I ran into this during `fprime-util build` of /GpsApp :
```
...
Scanning dependencies of target GpsApp
[100%] Building CXX object CMakeFiles/GpsApp.dir/Top/Main.cpp.o
In file included from /home/djwait/02_Projects/fprime/Drv/LinuxSerialDriver/LinuxSerialDriver.hpp:9,
                 from /home/djwait/02_Projects/fprime/GpsApp/build-fprime-automatic-native/GpsApp/Top/GpsAppTopologyAc.hpp:16,
                 from /home/djwait/02_Projects/fprime/GpsApp/Top/Main.cpp:6:
/home/djwait/02_Projects/fprime/Drv/LinuxSerialDriver/LinuxSerialDriverComponentImpl.hpp:74:65: error: conversion to ‘NATIVE_INT_TYPE’ {aka ‘int’} from ‘NATIVE_UINT_TYPE’ {aka ‘unsigned int’} may change the sign of the result [-Werror=sign-conversion]
   74 |       void startReadThread(NATIVE_INT_TYPE priority = Os::Task::TASK_DEFAULT, NATIVE_INT_TYPE stackSize = Os::Task::TASK_DEFAULT, NATIVE_INT_TYPE cpuAffinity = Os::Task::TASK_DEFAULT);
      |        
...
```
There were three of the same error, all on the same line (74):
```
...
 //! start the serial poll thread.
      //! buffSize is the max receive buffer size
      //!
      void startReadThread(NATIVE_INT_TYPE priority = Os::Task::TASK_DEFAULT, NATIVE_INT_TYPE stackSize = Os::Task::TASK_DEFAULT, NATIVE_INT_TYPE cpuAffinity = Os::Task::TASK_DEFAULT);
...
```
 I see that the three Os::Task::TASK_DEFAULT are NATIVE_UINT_TYPE , so I worked around that by changing all the `NATIVE_INT_TYPE` to `NATIVE_UINT_TYPE` ; not sure if that was the right thing to do, but the GpsApp was able to complete building.

Should I have done something else? Did I goof up something else?

**Comment by LeStarch** [comment, marked as answer]
Yes, it looks like this is the correct fix. This also looks like a bug in F´  in that those should be updated to the correct types.

**Reply by danjwait** [reply]
@LeStarch I made that change in my fork of devel, do you want me to make a pull request?

**Reply by LeStarch** [reply]
That would be awesome!

**Reply by danjwait** [reply]
Done! #1346 Thanks! Please let me know if there's anything else I can do to help / do differently next time
